expr/functions/filterSeries/function.go

```diff
@@ -0,0 +1,172 @@
+package filterSeries
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/bookingcom/carbonapi/expr/helper"
+	"github.com/bookingcom/carbonapi/expr/interfaces"
+	"github.com/bookingcom/carbonapi/expr/types"
+	"github.com/bookingcom/carbonapi/pkg/parser"
+)
+
+type filterSeries struct {
+	interfaces.FunctionBase
+}
+
+func GetOrder() interfaces.Order {
+	return interfaces.Any
+}
+
+func New(configFile string) []interfaces.FunctionMetadata {
+	f := &filterSeries{}
+	res := make([]interfaces.FunctionMetadata, 0)
+	for _, n := range []string{"filterSeries"} {
+		res = append(res, interfaces.FunctionMetadata{Name: n, F: f})
+	}
+	return res
+}
+
+var operators = map[string]struct{}{
+	"=":  struct{}{},
+	"!=": struct{}{},
+	">":  struct{}{},
+	">=": struct{}{},
+	"<":  struct{}{},
+	"<=": struct{}{},
+}
+
+// filterSeries(*seriesLists)
+func (f *filterSeries) Do(e parser.Expr, from, until int32, values map[parser.MetricRequest][]*types.MetricData) ([]*types.MetricData, error) {
+	args, err := helper.GetSeriesArg(e.Args()[0], from, until, values)
+	if err != nil {
+		return nil, err
+	}
+
+	callback, err := e.GetStringArg(1)
+	if err != nil {
+		return nil, err
+	}
+
+	var callbackFunc func([]float64, []bool) (float64, bool)
+	switch callback {
+	case "max":
+		callbackFunc = types.AggMax
+	case "min":
+		callbackFunc = types.AggMin
+	case "sum":
+		callbackFunc = types.AggSum
+	case "average":
+		callbackFunc = types.AggMean
+	case "last":
+		callbackFunc = types.AggLast
+	// TODO: this implementation does not support diff, median, multiply, range, stddev
+	default:
+		return nil, errors.New(fmt.Sprintf("unsupported consolidation function %v", callback))
+	}
+
+	operator, err := e.GetStringArg(2)
+	if err != nil {
+		return nil, err
```

Comment:
Add wrapping for more context on the top level.

Reply:
Wrapping added.